### PR TITLE
Intermediate certificate support

### PR DESCRIPTION
**Pull request by alex-wood**

### CODE REVIEW

File: lib/onelogin/ruby-saml/metadata.rb [hunk 1]
```diff
@@ -32,21 +32,20 @@ def generate(settings, pretty_print=false)
...
+          sp_cert_text = get_cert_text(cert)
+
+          add_sp_cert(sp_sso, "signing", sp_cert_text)
+          add_sp_cert(sp_sso, "encryption", sp_cert_text)
+
+          intermediate_certs = settings.get_sp_intermediate_certs
+          if intermediate_certs
+            intermediate_certs.each do |intermediate_cert|
+              intermediate_cert_text = get_cert_text(intermediate_cert)
+              add_sp_cert(sp_sso, "encryption", intermediate_cert_text)
```
Comment (Umofomia) [inline lib/onelogin/ruby-saml/metadata.rb]: You need to add these for both encryption and signing.

Comment (Umofomia) [inline lib/onelogin/ruby-saml/metadata.rb]: Also, you need to add the cert to the existing `ds:X509Data` element in each area. What you're doing here is creating new ones.

File: lib/onelogin/ruby-saml/settings.rb [hunk 2]
```diff
@@ -119,17 +120,26 @@ def get_fingerprint
...
+        get_cert(certificate)
+      end
+
+      def get_sp_intermediate_certs
+        return nil if intermediate_certs.nil? || intermediate_certs.empty?
+
+        intermediate_certs = []
+        idp_intermediate_certs.each do |intermediate_cert|
+          intermediate_certs << get_cert(intermediate_cert)
+        end
```
Comment (Umofomia) [inline lib/onelogin/ruby-saml/settings.rb]: Use `map` instead of `each` and `<<`.

File: lib/onelogin/ruby-saml/metadata.rb (hunk 1 above)
Comment (Umofomia) [inline lib/onelogin/ruby-saml/metadata.rb]: In other words, `intermediate_cert_texts` should be an additional option to the two calls to `add_sp_cert` above.

**Comment by alex-wood** [comment]
https://github.com/onelogin/ruby-saml/pull/324

**Pull request closed without merging**